### ISSUE TITLE
[BUG] mod() drops the gradient with respect to its divisor

### ISSUE THREAD

**Issue by christophercrouzet** [christophercrouzet/warp#13]
## What

For Warp's documented truncated-division modulo, `mod(a, b) = a - trunc(a / b) * b`. Its divisor derivative is `-trunc(a / b)` away from discontinuities, but Warp's adjoint only updates `adj_a`; `adj_b` remains zero.

## Reproduction

```python
import warp as wp

@wp.kernel
def f(a: wp.array[wp.float32], b: wp.array[wp.float32], y: wp.array[wp.float32]):
    y[0] = wp.mod(a[0], b[0])

for device in ("cpu", "cuda:0"):
    a = wp.array([3.5], dtype=wp.float32, requires_grad=True, device=device)
    b = wp.array([2.0], dtype=wp.float32, requires_grad=True, device=device)
    y = wp.zeros(1, dtype=wp.float32, requires_grad=True, device=device)
    with wp.Tape() as tape:
        wp.launch(f, dim=1, inputs=[a, b], outputs=[y], device=device)
    tape.backward(y)
    print(device, y.numpy(), a.grad.numpy(), b.grad.numpy())
```

Observed: value `1.5`, `d/da = 1`, `d/db = 0`. The analytic and finite-difference divisor gradient is `-1`.

## Why / when it matters

Optimizing or learning a modulus/divisor parameter produces a silently incomplete gradient for scalar and component-wise vector overloads.

## Environment

Reproduced against NVIDIA/warp `main` at `b1922eba85bd32af5e393c829c0b22e70ea7abea` (Warp 1.18.0.dev5) on Linux.

Reproduced on CPU and CUDA.
